.expeditor/verify.pipeline.yml

```diff
@@ -69,7 +69,9 @@ steps:
 
   - label: "[unit] applications-service"
     command:
-      - hab studio run "source .studiorc && go_component_unit applications-service && go_component_static_tests applications-service"
```

Comment:
I have the same question @afiune does. Are we moving away from using the hab studio for development or toward it? If we are moving toward using the studio more we should keep this test using the studio. 

From the poll that ssd did a few weeks ago, it looks like we most if not all people are using hab studio as their dev environment. 
![Screen Shot 2019-04-19 at 4 35 46 PM](https://user-images.githubusercontent.com/1679247/56448095-5ec9bb80-62c1-11e9-9e5f-691a6a92c199.png)

Reply:
> Are we moving away from using the hab studio for development or toward it?

I don't understand this question in the context of this pull request. As I wrote in the description, I did a bunch of work already in https://github.com/chef/automate/pull/110 to ensure you can use this both inside and outside of the studio as you like. Does it have to have a wrapper function in .studio to count as the studio?

As for that poll, I'm not sure we're measuring the same things. I personally never run any automate services locally, but I very frequently run static analysis and unit tests on my Mac since it's faster; I would probably say I "use the studio" as "my dev environment." I am pretty sure that's how people working on the other projects using Makefile.common_go are working since it was broken in the studio until I fixed it yesterday. But in any case, it works in the studio now, so you don't have to change how you do development at all, except that the specific procedure for running static tests in the studio has changed for this one component.

Reply:
@danielsdeleo I think the main problem is that you are switching the Buildkite test from running **_inside the studio_** to run **_outside with a Makefile_** that, as the poll says clearly, we are not running almost anything outside the studio. 

I think this PR would be perfectly fine if we switch the tests to run inside the studio as they were before. That is with `go_component_static_tests` and `go_componen_unit` methods.

Reply:
FWIW I think that poll likely isn't too informative on this point since it only provided one option.  For instance, I use the studio but almost always run the linter outside of the studio.

Right now, every project that uses the linter does so using a Makefile.

I think Dan did a pretty good job of mentioning the various things we might want to fix to move everything to running the linter inside the studio using a helper function:

- Apply the linter to ingest-service
- Apply the linter to config-mgmt-service (https://github.com/chef/automate/pull/127)
- Apply the linter to secrets-service (https://github.com/chef/automate/pull/127)
- Provide some means for `go_component_static_tests` to continue to allow components to specify their own linter args or fix all of the issues that led to custom exceptions:

```
components/automate-cli/Makefile:LINTERARGS:=--exclude-use-default -D lll ./...
components/automate-deployment/Makefile:LINTERARGS=--exclude-use-default -D lll ./...
components/compliance-service/Makefile:LINTERARGS=-v -D lll -D unparam -D nakedret -D prealloc --deadline 1m ./...
components/es-sidecar-service/Makefile:LINTERARGS=-D lll ./... ../../api/config/es-sidecar-service/...
components/nodemanager-service/Makefile:LINTERARGS=-v -D lll -D unparam -D nakedret -D prealloc --deadline 1m ./...
```

- Provide some means to continue to run the lint tests outside of the studio since most Go tools are 6 to 7 times studio inside Docker for Mac.

Personally, I would like to see us converge on something where every project had a standard `Makefile` that worked inside the studio and (for some tasks) outside the studio easily.  But it is a long road from where we are to that.

I think y'all should do whatever you come to agreement on for the application-service, but perhaps we can open an issue to track the above?

Reply:
I agree with @stevendanna — we should recognize all the amazing work that @danielsdeleo did, as usual, great work!

Having said that, I have never used a single make command since the creation of this repo, and even before that, since the repo was actually multiple repositories (before the monorepo) — I always run everything inside the studio and I would expect the CI system to do the same, that way we all have the same tools, compilers and versions. 

Now about the question that @lancewf asked, I am curious to know if this is a standard (to run the tests and linter outside the studio) and if it is, communicate it to everyone so that we all do the same. What I’m trying to avoid is maintaining multiple ways to develop and test. 

I hope this information clarified a bit more the situation.

Reply:
@danielsdeleo You are right. I don't see anything wrong with this PR. My question is more in general for Automate: which way are we moving towards? As @afiune said before, the old config-mgmt team always had everything run in the studio. Our reason for this was that it was easier to know that all the developers were using the same development environments (@afiune did a presentation on this a year back). This allowed us to assume if one person had a problem then we should see the same problem in our environment.  I know that not all the other teams have this same goal. Now we are the super team and a lot of the compliance tests and functions can only run outside of the studio. We are trying to move toward using the studio, but it is not a priority. 

It looks like you have answered my original question. We want to have a mix of development environments. Meaning people work outside of the studio and inside. Let me know if that is not correct.

Reply:
@lancewf my preference would be that everything works inside the studio, and some stuff optionally also works outside. The docker Mac filesystem is pretty slow so I personally find it nice to run these quick kinds of checks outside of the studio. In any case I added a helper function to `cd` to the `GOPATH` and run `make` for you and made that the ci script for this service. This way we can support customization of the linter for each service/team's needs with the makefile and have a standardized way to run it which gets tested in Ci.

Reply:
@danielsdeleo Thank you for hearing me out. This is great to hear. I think this is a great addition. I am finishing looking over the code now.